plugins/pluginmanagers/lazy.nix

```diff
@@ -0,0 +1,187 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib; let
+  cfg = config.plugins.lazy;
+  lazyPlugins = cfg.plugins;
+
+  processPlugin = plugin: let
+    mkEntryFromDrv = p:
+      if lib.isDerivation p
+      then {
+        name = "${lib.getName p}";
+        path = p;
+      }
+      else {
+        name = "${lib.getName p.pkg}";
+        path = p.pkg;
+      };
+    processDependencies =
+      if plugin ? dependencies && plugin.dependencies != null
+      then builtins.concatMap processPlugin plugin.dependencies
+      else [];
+  in
+    [(mkEntryFromDrv plugin)] ++ processDependencies;
+
+  processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
+  lazyPath = pkgs.linkFarm "lazy-plugins" processedPlugins;
+in {
+  options = {
+    plugins.lazy = {
+      enable = mkEnableOption "lazy.nvim";
+
+      plugins = with types; let
+        pluginType =
+          either
+          package
+          (submodule {
+            options = {
+              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
+
+              pkg = mkOption {
+                type = package;
+                description = "Vim plugin to install";
+              };
+
+              name = helpers.mkNullOrOption str "Name of the plugin to install";
+
+              dev = helpers.defaultNullOpts.mkBool false "When true, a local plugin directory will be used instead. See config.dev";
+
+              lazy = helpers.defaultNullOpts.mkBool true "When true, the plugin will only be loaded when needed. Lazy-loaded plugins are automatically loaded when their Lua modules are required, or when one of the lazy-loading handlers triggers";
+
+              enabled = helpers.defaultNullOpts.mkBool true "When false then this plugin will not be included in the spec";
+
+              cond =
```

Comment:
Could you use helpers.defaultNullOpts.mkStrLuaFnOr types.bool and metion it can take a `fun(LazyPlugin):boolean`

Reply:
```
cond = helpers.defaultNullOpts.mkStrLuaFnOr types.bool "When false, or if the function returns false, then this plugin will not be loaded. Useful to disable some plugins in vscode, or firenvim for example.";

```
results in:
```
 error: An option declaration for `plugins.lazy.plugins.*.cond' has type
       `lambda' rather than an attribute set.
       Did you mean to define this outside of `options'?

```

Any idea how I need to define this?